icefoganalytics/travel-authorization
# Build Out User Info Chip Component That Shows Full User Info

# Context

Convert the user's name field into a hoverable `v-chip` that on hover shows more detailed user info, the panel that gets opened should have an "edit" button that links to the user's profile page where they can update their info.

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/3095d2ab-f1bd-4b6b-9c65-864a0ea8634d)

Basically an effect like hovering on a github issue like this:
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/55171ed7-66bb-49fc-b0d5-01b2e2114fa8)

## Proposed Solution

Make "edit" button a link to the profile page instead. Defer profile edits to external user info service.

# Implementation

Add clickable user chip component that opens a menu with user details.
Contains link to user profile page.
Component is generic to user, but supports a "current-user" flag that customizes the user link.
Assume that user profiles are viewable by everyone, but that Auth0 subject (a.k.a. "sub") field is restricted information.
Only show content in user chip that has values set. e.g. if you don't have a mailcode, it doesn't show up in your chip.
Link the user's email to their gravatar.

# Testing Instructions

1. Boot the app via `dev up`
2. Log in to the app at http://localhost:8080/
3. Go to the My Travel Requests page via the top nave dropdown menu.
4. Click on a "Draft" travel authorization, or create one if none exist in the "My Travel" table.
5. Check that the header is now "Travel - `<current user chip>`".
6. Click on the user's full name to see their details.
7. Check that you can click the link icon and it takes you to the profile page.
